native/src/components/CitySelector.tsx

```diff
@@ -1,188 +1,85 @@
-
-const byNameAndAliases = (name: string) => (city: CityModel) =>
-  normalizeSearchString(city.name).includes(name) || checkAliases(city, name)
-
-class CitySelector extends React.PureComponent<PropsType> {
-  _filter(): Array<CityModel> {
-    const { cities, filterText } = this.props
-    const normalizedFilter = normalizeSearchString(filterText)
-
-    if (normalizedFilter === 'wirschaffendas') {
-      return cities.filter(_city => !_city.live)
-    }
-
-    return cities
-      .filter(_city => _city.live || buildConfig().featureFlags.developerFriendly)
-      .filter(byNameAndAliases(normalizedFilter))
-  }
-
-  // Landkreis should come before Stadt
-  _sort(cities: Array<CityModel>): Array<CityModel> {
-    return cities.sort((a, b) => {
-      // There is currently a bug in hermes crashing the app if using localeCompare on empty string
-      // Therefore the following does not work if there are two cities with the same sortingName of which one has no prefix set:
-      // return a.sortingName.localeCompare(b.sortingName) || (a.prefix || '').localeCompare(b.prefix || '')
-      // https://github.com/facebook/hermes/issues/602
-      const sortingNameCompare = a.sortingName.localeCompare(b.sortingName)
-      if (sortingNameCompare !== 0) {
-        return sortingNameCompare
-      }
-      if (!b.prefix) {
-        return 1
-      }
-      if (!a.prefix) {
-        return -1
-      }
-      return a.prefix.localeCompare(b.prefix)
-    })
-  }
-
-  _renderFilteredLocations(cities: Array<CityModel>): React.ReactNode {
-    const { theme, filterText, navigateToDashboard } = this.props
-    if (cities.length === 0) {
-      return <NothingFound paddingTop />
-    }
-    const sorted = this._sort(cities)
-    const groups = groupBy(sorted, (city: CityModel) => city.sortCategory)
-    return transform(
-      groups,
-      (result: React.ReactNode[], cities: CityModel[], key: string) => {
-        result.push(
-          <CityGroupContainer key={key}>
-            <CityGroup>{key}</CityGroup>
-            {cities.map(city => (
-              <CityEntry
-                key={city.code}
-                city={city}
-                filterText={filterText}
-                navigateToDashboard={navigateToDashboard}
-                theme={theme}
-              />
-            ))}
-          </CityGroupContainer>
-        )
-      },
-      []
-    )
-  }
-
-  _renderNearbyLocations(): React.ReactNode {
-    const { cities, t, theme, navigateToDashboard, filterText, locationInformation } = this.props
-    const { requestAndDetermineLocation } = locationInformation
-
-    if (locationInformation.coordinates) {
-      const [longitude, latitude] = locationInformation.coordinates
-      const nearbyCities = getNearbyCities(
-        cities.filter(city => city.live),
-        longitude,
-        latitude
+const CitySelector = ({ cities, navigateToDashboard, t }: PropsType): ReactElement => {
```

Comment:
please use `useTranslations` hook instead of passing it from `Landing.tsx`

Reply:
I can do that, what is the benefit of this though?

Reply:
not have to pass in the component and each test the `t` i mean we can pass from every root component the translation to the last child. so could avoid hooks at all :)
Even you don't have to do that

Reply:
You're right, will change :)